magicctapipe/io/io.py

```diff
@@ -908,28 +737,41 @@ def load_mc_dl2_data_file(input_file, quality_cuts, event_type, weight_type_dl2)
         If the input event type is not known
     """
 
+    TEL_NAMES, TEL_COMBINATIONS = telescope_combinations(config)
+    combo_types = np.asarray(range(len(TEL_COMBINATIONS)))
+    three_or_more = []
+    for n, combination in enumerate(TEL_COMBINATIONS.values()):
+        if len(combination) >= 3:
+            three_or_more.append(n)
+
     # Load the input file
     df_events = pd.read_hdf(input_file, key="events/parameters")
     df_events.set_index(["obs_id", "event_id", "tel_id"], inplace=True)
     df_events.sort_index(inplace=True)
 
-    df_events = get_stereo_events_old(df_events, quality_cuts)
+    df_events = get_stereo_events(
+        df_events, config, quality_cuts, eval_multi_combo=False
+    )
 
     logger.info(f"\nExtracting the events of the '{event_type}' type...")
 
     if event_type == "software":
         # The events of the MAGIC-stereo combination are excluded
-        df_events.query("(combo_type > 0) & (magic_stereo == True)", inplace=True)
+        df_events.query(
+            f"(combo_type < {combo_types[-1]}) & (magic_stereo == True)", inplace=True
+        )
+
+    elif event_type == "software_3tels_or_more":
+        df_events.query(f"combo_type == {three_or_more}", inplace=True)
 
-    elif event_type == "software_only_3tel":
-        df_events.query("combo_type == 3", inplace=True)
+    elif event_type == "software_6_tel":
```

Comment:
the different event types should be revised or at least described more clearly.
As far as I understand in the new scheme: 
"software" requires that MAGIC triggered the event but events without any LST we exclude (this is consistent with the old scheme)
"software_3tels_or_more" takes any combination of 3 telescopes, but can accept e.g. M1+LST1+LST2, and such a combination is only possible in hardware and not software scheme.
"software_6_tel" - this seem to accept any combination except of M1+M2, and it also does not check if MAGIC alone triggered, so also e.g. M1+LST1+LST2 would trigger this, which is neither 6 tel, nor software. 

of course whatever is changed here, will need to be changed for data as well.

Reply:
@jsitarek. I tried to fix them a bit in MCs; if they are OK, I will move them to a separate function, fix data and doc. Thanks!

Reply:
I checked the new names and I think they are fine. "magic-only" would really mean only events that were triggered by only MAGIC and none of LST1 telescopes, but this kind of analysis only makes sense if you want to analyze MAGIC only data so the LSTs either way need to be removed from the chain (but then they need to be removed at an earlier stage)

Reply:
Ok thank you. I will fix the data ones too.
MAGIC-only must be requested since the stereo script, where, in case of magic-only, only the two MAGIC images are used (as it is now). Moreover, I think this kind of analysis will be used only with MAGIC+LST1 (to compare with MARS) and is a bit meaningless when we use 4 LSTs

Reply:
fixed